src/components/CertificateDropZone/CertificateDropZone.tsx

```diff
@@ -30,16 +32,22 @@ export const CertificateDropZone: FunctionComponent<CertificateDropzoneProps> =
     dispatch(resetCertificateState());
   }, [dispatch]);
 
+  const { currentChainId, changeNetwork } = useProviderContext();
+
   const onDrop = useCallback(
     (acceptedFiles: Blob[]) => {
       acceptedFiles.forEach((file: Blob) => {
         const reader = new FileReader();
 
         reader.onabort = () => console.log("file reading was aborted");
         reader.onerror = () => console.log("file reading has failed");
-        reader.onload = () => {
+        reader.onload = async () => {
           try {
             const json = JSON.parse(reader.result as string);
+            const chainId = getChainId(json);
+            if (chainId && currentChainId !== chainId) {
+              await changeNetwork(chainId);
```

Comment:
i think this conflicts with the flow for #550

Reply:
<img width="887" alt="image" src="https://user-images.githubusercontent.com/68947167/155951846-59f92090-b6d8-461c-9ac7-8268ef40e230.png">